### PR TITLE
Supermatter alerts now only play on the Z level the supermatter is on

### PR DESCRIPTION
**Pull request by ghost**
No more 'Oh fuck call the shuttle and ruin our engineers hard work' because a bored MoMMI accidentally sharded themselves, causing a thermal runaway reaction with the shard. Kinda makes sense considering supermatter no longer affects things not on their Z level as of #8285

Tested a fair bit, but you can only get so far with only being able to be in one place at a time

Not sure if it's associated with this, but I was getting a lot of 

`visible_message is attempting to call on_see on a hearer that isn't attached to anything: . Previous type of attached: /obj/item/device/radio/headset.`

while testing this. Might be Broadcast_message related?

### CODE REVIEW

**Comment by Exxion** [comment]
That is a known runtime with an unknown cause
Not related to this PR

(More accurately, it's debug output recently added to attempt to diagnose said runtime)

**Comment by ghost** [comment]
Ah okay, thanks for clarifying, I've been trying to hunt down what I coulda done to cause that

**Comment by ComicIronic** [comment]
Interesting that it's headsets, though. Might be an easy cleanup now that we know.

File: code/modules/supermatter/supermatter.dm [hunk 1]
```diff
@@ -193,16 +194,14 @@
 			speech.name = "Supermatter [short_name] Monitor"
 			speech.job = "Automated Announcement"
 			speech.as_name = "Supermatter [short_name] Monitor"
-			Broadcast_Message(speech, level = list(0,1))
+			Broadcast_Message(speech, level = current_zlevel)
```
Comment (ComicIronic) [inline code/modules/supermatter/supermatter.dm]: I'm legitimately surprised this works, but you should keep it as a list.

File: html/changelogs/skullyton.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# There needs to be a space after the - and before the prefix. Don't use tabs anywhere in this file.
+# Also, this gets changed to [] after reading.  Just remove the brackets when you add new shit.
+# If you're using characters such as # ' : - or the like in your changelog, surround the entire change with quotes as shown in the second example. These quotes don't show up on the changelog once it's merged and prevent errors.
+# SCREW ANY OF THIS UP AND IT WON'T WORK.
+changes: 
+- rscadd: Supermatter alerts now only play on the z level they are located on
+- rscdel: Fun
```
Comment (ComicIronic) [inline html/changelogs/skullyton.yml]: No memes.

Comment (ghost) [inline html/changelogs/skullyton.yml]: Now I have to remove fun by removing the fact I removed fun

File: code/modules/supermatter/supermatter.dm (hunk 1 above)
Comment (ghost) [inline code/modules/supermatter/supermatter.dm]: So level = list(current_zlevel)?

Makes sense, seemed a bit shaky when testing it

Comment (ComicIronic) [inline code/modules/supermatter/supermatter.dm]: Yes. The proc expects a list, and I think you got by with this thanks to some very shakey Byond internal logic to do with how `in` works.

**Comment by ComicIronic** [comment]
Also, make unique changelog files with delete after set to true to avoid conflicts.

**Comment by ghost** [comment]
It is set to true though

**Comment by ghost** [comment]
All's good now

**Comment by jknpj** [comment]
A direct buff to xenoarch antag. Now if they're luck then can blow up a crystal stealthly

**Pull request merged**